tools/ide_plugins/vscode/package.json

```diff
@@ -31,10 +31,11 @@
   "devDependencies": {
     "@types/mocha": "^2.2.44",
     "@types/node": "^8.0.53",
+    "@vscode/python-extension": "^1.0.5",
```

Comment:
is it also necessary to add an `extensionDependency` on this extension? my local testing shows the following:

```
2024-04-29 11:58:25.776 [error] Error: Cannot find module '@vscode/python-extension'
Require stack:
- /Users/kylei/.vscode/extensions/fb-pyre-check.pyre-vscode-0.0.2/out/main.js
- /private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/loader.js
- /private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/bootstrap-amd.js
- /private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/bootstrap-fork.js
	at Module._resolveFilename (node:internal/modules/cjs/loader:987:15)
	at Module._load (node:internal/modules/cjs/loader:832:27)
	at c._load (node:electron/js2c/asar_bundle:5:13343)
	at a._load (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:118:14563)
	at v._load (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:113:64900)
	at w._load (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:113:64268)
	at Module.require (node:internal/modules/cjs/loader:1059:19)
	at p (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/loader.js:4:699)
	at Object.<anonymous> (/Users/kylei/.vscode/extensions/fb-pyre-check.pyre-vscode-0.0.2/out/main.js:24:28)
	at u._compile (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/loader.js:4:1313)
	at Module._extensions..js (node:internal/modules/cjs/loader:1220:10)
	at Module.load (node:internal/modules/cjs/loader:1035:32)
	at Module._load (node:internal/modules/cjs/loader:876:12)
	at c._load (node:electron/js2c/asar_bundle:5:13343)
	at a._load (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:118:14563)
	at v._load (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:113:64900)
	at w._load (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:113:64268)
	at Module.require (node:internal/modules/cjs/loader:1059:19)
	at require (node:internal/modules/cjs/helpers:102:18)
	at Function.r [as __$__nodeRequire] (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/loader.js:5:101)
	at h._loadCommonJSModule (/private/var/folders/x0/z3j4vbzs3wqd0ynb6q24jglm0000gn/T/AppTranslocation/6187919E-1A75-43C1-BC3A-4E2581821C40/d/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:113:66459)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async Promise.all (index 0)
```

Reply:
Really sorry about this, I had only tested in debug mode.

The issue has been resolved in https://github.com/facebook/pyre-check/pull/820/commits/e9147b205f42ff43592adf67eae621696621df1e .

`extensionDependency` is not necessary as this should work with or without the VsCode Python dependency installed. What was needed is promoting the `@vscode/python-extension` from a `devDependency` to a `dependency`. This is needed because I'm using a string literal from the package (the name of the VsCode python dependency) that is actually used at runtime:

```
// Output JS from tsc

const python_extension_1 = require("@vscode/python-extension");
...
function activate(_) {
    return __awaiter(this, void 0, void 0, function* () {
        const pythonExtension = vscode.extensions.getExtension(python_extension_1.PVSC_EXTENSION_ID);
...
```

I tested by running:

```
npm run compile
vsce package
code --install-extension pyre-vscode-0.0.2.vsix
```

and it seems to work as expected.